src/Views/Me/AddBook.js

```diff
@@ -0,0 +1,124 @@
+import React, {useState} from 'react';
+import { useHistory } from "react-router-dom"
+
+const AddBook = () => {
+
+    const [titleInput, setTitleInput] = useState("");
+    const [authorInput, setAuthorInput] = useState("");
+    const [genresInput, setGeneresInput] = useState("");
+    const [releaseDateInput, setReleaseDateInput] = useState("");
+    const [publisherInput, setPublisherInput] = useState("");
+    const [descriptionInput, setDescriptionInput] = useState("");
+    const [creationStatus, setCreationStatus] = useState();
+    const history = useHistory();
+
+    const handleInputChange = (e) => {
+        if (e.target.id === "bookTitle") return setTitleInput(e.target.value);
+        if (e.target.id === "author") return setAuthorInput(e.target.value);
+        if (e.target.id === "genres") return setGeneresInput(e.target.value);
+        if (e.target.id === "releaseDate") return setReleaseDateInput(e.target.value);
+        if (e.target.id === "publisher") return setPublisherInput(e.target.value);
+        if (e.target.id === "description") return setDescriptionInput(e.target.value);
+    };
+
+    const handleSubmit = (e) => {
+        console.log('Send submit ...')
+        setCreationStatus("Request is being sent");
+        e.preventDefault();
+        addNewBook(titleInput, authorInput, genresInput, releaseDateInput, publisherInput, descriptionInput );
+        setTitleInput("");
+        setAuthorInput("");
+        setGeneresInput("");
+        setReleaseDateInput("");
+        setPublisherInput("");
+        setDescriptionInput("");
+    };
+
+    const addNewBook = async (title, author, genres, releaseDate, publisher, description) => {
+        let res = await fetch("https://pikabook.herokuapp.com/api/books", {
+          method: "post",
+          headers: {
+            Accept: "application/json, text/plain, */*",
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            title,
+            author,
+            genres,
+            releaseDate,
+            publisher,
+            description
+          }),
+        });
+        if (!res.ok) {
+          res = await res.text();
+          setCreationStatus(res);
+        } else {
+          res = await res.json();
+          setCreationStatus("Book successfully added");
+          setTimeout(() => {
+            history.push("/me")
+          }, 2000)
+        }
+    };
```

Comment:
I think that logic associated with posting data shouldn't lay in React components. Better place for such code would be some external file (i.e. called bookService.jsx) to increase readability. What do you think about it?

Reply:
Yes, I think it is good idea. I need to think how to implement it. But on the other hand I think that components in this project contains logic ex. Registration.js. Should it also be changed?